In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

In [2]:
# Importing data
df = pd.read_csv(r'C:\Users\Lore\12.2020ProiectCursDA\6 Covid Economies\covid economies.csv', index_col = False)

In [3]:
path = r'C:\Users\Lore\12.2020ProiectCursDA'

In [4]:
path

'C:\\Users\\Lore\\12.2020ProiectCursDA'

In [5]:
#Q7
df.head()

CODE      COUNTRY   CITY CONTINENT  EU        DATE    HDI   TC   TD  STI  \
0  AFG  Afghanistan  Kabul      Asia  No  12/31/2019  0.498  0.0  0.0  0.0   
1  AFG  Afghanistan  Kabul      Asia  No    1/1/2020  0.498  0.0  0.0  0.0   
2  AFG  Afghanistan  Kabul      Asia  No    1/2/2020  0.498  0.0  0.0  0.0   
3  AFG  Afghanistan  Kabul      Asia  No    1/3/2020  0.498  0.0  0.0  0.0   
4  AFG  Afghanistan  Kabul      Asia  No    1/4/2020  0.498  0.0  0.0  0.0   

   Population    GDPCAP  
0   17.477233  7.497754  
1   17.477233  7.497754  
2   17.477233  7.497754  
3   17.477233  7.497754  
4   17.477233  7.497754

In [6]:
#Q7
df.shape

(49991, 12)

In [7]:
#Q7
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49991 entries, 0 to 49990
Data columns (total 12 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   CODE        49991 non-null  object 
 1   COUNTRY     49991 non-null  object 
 2   CITY        49991 non-null  object 
 3   CONTINENT   49991 non-null  object 
 4   EU          49991 non-null  object 
 5   DATE        49991 non-null  object 
 6   HDI         44216 non-null  float64
 7   TC          49991 non-null  float64
 8   TD          49991 non-null  float64
 9   STI         49991 non-null  float64
 10  Population  49991 non-null  float64
 11  GDPCAP      49991 non-null  float64
dtypes: float64(6), object(6)
memory usage: 4.6+ MB


The data set has 6 variable of float type and 6 type of object.

In [8]:
#Q7
df.dtypes

CODE           object
COUNTRY        object
CITY           object
CONTINENT      object
EU             object
DATE           object
HDI           float64
TC            float64
TD            float64
STI           float64
Population    float64
GDPCAP        float64
dtype: object

In [10]:
#Q7
df.columns

Index(['CODE', 'COUNTRY', 'CITY', 'CONTINENT', 'EU', 'DATE', 'HDI', 'TC', 'TD',
       'STI', 'Population', 'GDPCAP'],
      dtype='object')

In [9]:
#Q7
df.describe()

HDI            TC            TD           STI    Population  \
count  44216.000000  49991.000000  49991.000000  49991.000000  49991.000000   
mean       0.720139      6.789750      3.431823      3.206050     15.484868   
std        0.160902      3.636774      3.087061      1.654480      2.462087   
min        0.000000      0.000000      0.000000      0.000000      6.695799   
25%        0.601000      4.189655      0.000000      2.967333     14.347070   
50%        0.752000      7.132498      3.218876      4.008968     15.973604   
75%        0.847000      9.522959      5.641907      4.341855     17.210890   
max        0.953000     15.914092     12.299900      4.605170     21.087439   

             GDPCAP  
count  49991.000000  
mean       8.342164  
std        3.143257  
min        0.000000  
25%        7.971401  
50%        9.368531  
75%       10.229832  
max       11.669379

From this statistical describe I can see that all the variables are good. No outliers.

In [10]:
#Q6
#Finding Missing Values
df.isnull().sum()

CODE             0
COUNTRY          0
CITY             0
CONTINENT        0
EU               0
DATE             0
HDI           5775
TC               0
TD               0
STI              0
Population       0
GDPCAP           0
dtype: int64

In HDI column are 5775 missing values. There is an explanation: This is a composite measure based on nine response indicators and because this analysis is for all the countries, the pours countries don't have information available.

In [11]:
#Q6
#Replace missing values
df['HDI'].fillna(0, inplace=True)

In [12]:
df.isnull().sum()

CODE          0
COUNTRY       0
CITY          0
CONTINENT     0
EU            0
DATE          0
HDI           0
TC            0
TD            0
STI           0
Population    0
GDPCAP        0
dtype: int64

In [13]:
#Q6
#look for full duplicates
df_dups = df[df.duplicated()]

In [14]:
df_dups

Empty DataFrame
Columns: [CODE, COUNTRY, CITY, CONTINENT, EU, DATE, HDI, TC, TD, STI, Population, GDPCAP]
Index: []

No duplicates.

In [15]:
df.shape

(49991, 12)

In [16]:
df.to_csv(os.path.join(path, '6 Covid Economies', 'df.csv'))